### ISSUE TITLE
Swagger-Parser state

### ISSUE THREAD

**Issue by jmini** [OpenAPITools/openapi-generator#68]
This issue is to discuss the state of one of our upstream dependency `Swagger-Parser`.

Original project: https://github.com/swagger-api/swagger-parser/pulls

Our fork: https://github.com/OpenAPITools/swagger-parser/pulls
In our fork, we have started to do some fixes on the [2.0-OpenAPITools](https://github.com/OpenAPITools/swagger-parser/tree/2.0-OpenAPITools) branch.

---
Personally I would prefer to have the issues fixed in the original project. Otherwise this means that we will have to publish a release of `OpenAPITools/swagger-parser` under new maven coordinates (using our groupId)

I would prefer not fragmenting the `swagger-parser` project.

I am using other tools like [openapi-diff](https://github.com/quen2404/openapi-diff) and I would like all project to use the same version of `swagger-parser`. Otherwise this means that we will have a dependency management hell.

---

I propose that we still open issue and file PRs in the original `swagger-parser` project. I know there is little reaction at the moment, but it would be better on the long term.

We can also add all the PRs we have created on the original project to our `2.0-OpenAPITools`

If they do not integrate our fixes we will be forced to create a release of `org.openapitools:swagger-parser` based on the `2.0-OpenAPITools`. But I hope this will not be necessary.

**Comment by jmini** [comment]
PR that we do not manage to get in:
* https://github.com/swagger-api/swagger-parser/pull/696 by @ackintosh 

I have also opened two issues this morning (I did not investigate them yet):
* https://github.com/swagger-api/swagger-parser/issues/708
* https://github.com/swagger-api/swagger-parser/issues/709

**Comment by jmini** [comment]
@frantuma thank you for this reply! This is encouraging.

_Please notice that I can only speak for myself, and other members of this community might have other opinion._

I will do my best to make this project continue to use the official version of Swagger-Parser. In my opinion this requires:

* Quick feedback (merge, review or wontfix explanation) on PRs. => couple of days seems reasonable.
* A clear process to ask for a release. I do not want any special treatment, but it seems reasonable to do releases from time to time. => this is important for all downstream projects, not just for this one.

I do not see a reason to publish an alternative release or to use https://jitpack.io/ if the described mindset is present. At the end I think that we can all work together...

**Comment by frantuma** [comment]
@jmini thanks for your feedback. As for your second point we are considering this and will get back to you and the community.

**Comment by jmini** [comment]
Just for tracking:

* swagger-api/swagger-parser#710 is needed for #67

**Comment by jmini** [comment]
2018-05-17: both branches `2.0-OpenAPITools` and `2.0` are aligned. No pending PR from our side.
Thank you to everyone involved. We keep this issue open for additional discussion.

One point needs to be answered:
> @jmini thanks for your feedback. As for your second point (a clear process to ask for a release) we are considering this and will get back to you and the community.

**Comment by jmini** [comment]
(tracking)

PR:

* https://github.com/swagger-api/swagger-parser/pull/716

**Comment by jmini** [comment]
@frantuma: I took the liberty to open https://github.com/swagger-api/swagger-parser/issues/717 to discuss it. I am not sure it is the appropriate process.

**Comment by jmini** [comment]
Dear Swagger-Parser maintainers,

I have noticed a lot of activity around the project with great patches contributed over the few days. Some of them are patches for issues that were raised by OpenAPI-Generator users.

We also got one patch submitted directly to our fork. I have asked the contributor to also contribute its change against the official project.

I have started to review some of them. During my review I have suggested to add Unit Tests.

I think it would be great to have them integrated into the official project. Let me know what I can do additionally to help you in this task.

---

In Parallel to the review I have also integrated the patch in the alternative branch (`2.0-OpenAPITools`) we maintain in the [`OpenAPITools/swagger-parser`](https://github.com/OpenAPITools/swagger-parser/) fork.

This allows to perform some tests.

---

I will try to keep a list up to date of the changes:
(If you follow the same review and merge order, you will be able to have the same conflict resolution I did on the `2.0-OpenAPITools` branch)

* swagger-api/swagger-parser#753
* swagger-api/swagger-parser#759
* swagger-api/swagger-parser#763
* ...
* _more to come_

cc: @frantuma

**Comment by jmini** [comment]
`2.0.2` has been released but contains a regression that prevent us to use this version. See #787 for more detail. We will stay to [`2.0.2-OpenAPITools.org-1`](https://github.com/OpenAPITools/swagger-parser/releases/tag/v2.0.2-OpenAPITools.org-1) until a `2.0.3` is published.

---

I track here the changes we are interested in for a `2.0.3` release. We might come up with a `2.0.3-OpenAPITools.org-1` for our `3.3.0` release (due to end August 2018 / first week of September 2018).

* Fix for https://github.com/swagger-api/swagger-parser/issues/799 (or as workaround stay on `1.0.36` until `1.0.38` is published -- we do not want to rely on SNAPSHOT versions in OpenAPI-Generator project)
* https://github.com/swagger-api/swagger-parser/pull/806 (fix for issue https://github.com/swagger-api/swagger-parser/pull/805)

**Comment by jmini** [comment]
Version [2.0.3-OpenAPITools.org-1](https://github.com/OpenAPITools/swagger-parser/releases/tag/v2.0.3-OpenAPITools.org-1) of Swagger-Parser was published on maven central:

```xml
<dependency>
  <groupId>org.openapitools.swagger.parser</groupId>
  <artifactId>swagger-parser</artifactId>
  <version>2.0.3-OpenAPITools.org-1</version>
</dependency>
```

---

In https://github.com/swagger-api/swagger-parser/pull/806#issuecomment-415559341, the timeline for `2.0.3` was given:

> hi @douglasbgray the current plan is to release in a couple of weeks.

This might be to late for our `3.3.0` release. So we will use `2.0.3-OpenAPITools.org-1` until the official version is published. => Change made with PR https://github.com/OpenAPITools/openapi-generator/pull/951

**Comment by jmini** [comment]
I track here the changes we are interested in for a `2.0.6` release.

* https://github.com/swagger-api/swagger-parser/pull/880 (fix for issue https://github.com/swagger-api/swagger-parser/issues/879)

**Comment by jmini** [comment]
I track here the changes we are interested in for a `2.0.8` release.

* https://github.com/swagger-api/swagger-parser/pull/969 (fix for issue https://github.com/swagger-api/swagger-parser/issues/968)
* https://github.com/swagger-api/swagger-parser/pull/967 (fix for issue https://github.com/swagger-api/swagger-parser/issues/966)
